src/backend/distributed/sql/udfs/citus_drop_trigger/latest.sql

```diff
@@ -22,7 +17,19 @@ BEGIN
         -- then drop all the shards on the workers
         -- finally remove the pg_dist_partition entry on the coordinator
         PERFORM master_remove_distributed_table_metadata_from_workers(v_obj.objid, v_obj.schema_name, v_obj.object_name);
-        PERFORM citus_drop_all_shards(v_obj.objid, v_obj.schema_name, v_obj.object_name);
+
+        -- If both original and normal values are false, the table was a partition
+        -- its shards will be dropped when dropping the parent's shards,
+        -- so we can skip: i.e. we call citus_drop_all_shards with drop_shards_metadata_only parameteter set to true
+        -- NOTE: the other way around is not true:
```

Comment:
minor: can we use local variables/parameters such as (syntax  is wrong) to improve the readability a little more:
```
droppedTableIsAPartition := v_obj.original = false AND v_obj.normal = false

...
citus_drop_all_shards(v_obj.objid, v_obj.schema_name, v_obj.object_name, dropOnlyShardMetadata: true);
...

```

Reply:
The problem with a name like `droppedTableIsAPartition` is that it gives the idea that this is an "if and only if" condition, which is not the case.

Reply:
> The problem with a name like droppedTableIsAPartition is that it gives the idea that this is an "if and only if" condition, which is not the case.

I don't think it gives the `if and only if` condition with such a name. The problem is that it requires substantial effort for readers to understand what is going on. Not many people know the meanings of `normal` and `original` params. I think a variable can make the readability much simpler, but I'll let you decide.

Reply:
> I don't think it gives the if and only if condition with such a name.

As long as that's not the case, a parameter could simplify things. Thanks.